projects/bb3/agents/opt_bb3_agent.py

```diff
@@ -701,7 +725,15 @@ def _failed_messages(replies):
             retries += 1
             n_mems = [min(1, len(obs['memories']) // 3) for obs in opening_obs]
             for i, o in enumerate(opening_obs):
-                o.force_set('memories', random.sample(o['memories'], n_mems[i]))
+                mem_indices = random.sample(range(len(o['memories'])), n_mems[i])
+                o.force_set(
```

Comment:
Does this purge the existing memories if the current item in the batch doesn't have memory access this round?

Reply:
no, we're just playing around with the memories sent in the opening message. This is legacy logic for if the opening message is not meaningful (back when our models couldn't handle the opener)